commit dbed4a797e00d98a4eaa971f1585091fad076a74
Author: Jocelyn Falempe <jfalempe@redhat.com>
Date:   Tue Jan 28 17:52:10 2025 +0100

    drm/panic: Better binary encoding in QR code
    
    The current encoding, is done by converting 13bits of input into 4
    decimal digits, that are then encoded efficiently using the numeric
    encoding of the QR code specification.
    The Fido v2.2 specification [1] uses a similar approach for its
    QR-initiated authentication. The only difference is that it converts
    7 bytes (56bits) of input into 17 decimal digits. The benefit is that
    the algorithm doesn't require to split input bytes into 13bits chunk,
    and the ratio is a bit better.
    This improvement was proposed by Jó Ágila Bitsch in [2].
    drm_panic is still young, and the QR code feature is not widely used,
    so it's still time to switch to a common algorithm, shared with a
    widely used standard.
    I also changed the name of the url parameter, from zl= to z=, so the
    website can keep backward compatibility if needed.
    
    [1] https://fidoalliance.org/specs/fido-v2.2-rd-20230321/fido-client-to-authenticator-protocol-v2.2-rd-20230321.html#hybrid-qr-initiated
    [2] https://github.com/kdj0c/panic_report/issues/2
    
    Signed-off-by: Jocelyn Falempe <jfalempe@redhat.com>
    Acked-by: Maxime Ripard <mripard@kernel.org>
    Link: https://patchwork.freedesktop.org/patch/msgid/20250128165254.893204-1-jfalempe@redhat.com

diff --git a/drivers/gpu/drm/drm_panic.c b/drivers/gpu/drm/drm_panic.c
index f128d345b16d..ab42a2b1567d 100644
--- a/drivers/gpu/drm/drm_panic.c
+++ b/drivers/gpu/drm/drm_panic.c
@@ -499,7 +499,7 @@ static int drm_panic_get_qr_code_url(u8 **qr_image)
 	char *kmsg;
 	int max_qr_data_size, url_len;
 
-	url_len = snprintf(url, sizeof(url), CONFIG_DRM_PANIC_SCREEN_QR_CODE_URL "?a=%s&v=%s&zl=",
+	url_len = snprintf(url, sizeof(url), CONFIG_DRM_PANIC_SCREEN_QR_CODE_URL "?a=%s&v=%s&z=",
 			   utsname()->machine, utsname()->release);
 
 	max_qr_data_size = drm_panic_qr_max_data_size(panic_qr_version, url_len);